src/EditorFeatures/Core/Remote/SolutionChecksumUpdater.cs

```diff
@@ -104,81 +93,58 @@ private void PauseWork()
         {
             // An expensive global operation started (like a build).  Pause ourselves and cancel any outstanding work in
             // progress to synchronize the solution.
-            //
-            // Note: We purposefully ignore the cancellation token produced by CreateNext.  The purpose here is to
-            // cancel the token controlling the current work, but not have any uncanceled token for new work to use when
-            // it comes in.
-            _pauseCancellationSeries.CreateNext();
+            lock (_gate)
+            {
+                _synchronizeWorkspaceQueue.CancelExistingWork();
+                _isPaused = true;
+            }
         }
 
         private void ResumeWork()
         {
-            // create a new token to control all the work we need to do while we're current unpaused.
-            var nextToken = _pauseCancellationSeries.CreateNext();
-            _currentWorkToken = nextToken;
-            // Start synchronizing again.
-            _synchronizeWorkspaceQueue.AddWork(nextToken);
+            lock (_gate)
+            {
+                _synchronizeWorkspaceQueue.AddWork();
+                _isPaused = false;
+            }
         }
 
         private void OnWorkspaceChanged(object? sender, WorkspaceChangeEventArgs e)
         {
             // Check if we're currently paused.  If so ignore this notification.  We don't want to any work in response
             // to whatever the workspace is doing.
-            var workToken = _currentWorkToken;
-            if (workToken.IsCancellationRequested)
-                return;
+            lock (_gate)
+            {
+                if (_isPaused)
+                    return;
+            }
 
             if (e.Kind == WorkspaceChangeKind.DocumentChanged)
             {
                 _textChangeQueue.AddWork((e.OldSolution.GetDocument(e.DocumentId), e.NewSolution.GetDocument(e.DocumentId)));
             }
 
-            _synchronizeWorkspaceQueue.AddWork(workToken);
+            _synchronizeWorkspaceQueue.AddWork();
```

Comment:
should this be passing `cancelExistingWork`?  same above

Reply:
IN this case, definite no.  The checksumupdater doesn't want to cancel items.  it wants them to run to completion *normally*.  It *does* cancel when it hears that it should be pausing work because of a 'BUILD' happening.